src/buildtool/catkin-make.ts

```diff
@@ -1,43 +1,49 @@
-// Copyright (c) Andrew Short. All rights reserved.
-// Licensed under the MIT License.
-
-import * as vscode from "vscode";
-
-import * as extension from "../extension";
-
-/**
- * Provides catkin build and test tasks.
- */
-export class CatkinTaskProvider implements vscode.TaskProvider {
-    public provideTasks(token?: vscode.CancellationToken): vscode.ProviderResult<vscode.Task[]> {
-        let buildCommand: string;
-        let testCommand: string;
-
-        if (extension.buildSystem === extension.BuildSystem.CatkinMake) {
-            buildCommand = `catkin_make --directory "${extension.baseDir}"`;
-            testCommand = `${buildCommand} run_tests`;
-        } else if (extension.buildSystem === extension.BuildSystem.CatkinTools) {
-            buildCommand = `catkin build --workspace "${extension.baseDir}"`;
-            testCommand = `${buildCommand} --catkin-make-args run_tests`;
-        }
-
-        const make = new vscode.Task({ type: "catkin" }, "make", "catkin");
-        make.execution = new vscode.ShellExecution(buildCommand, {
-            env: extension.env
-        });
-        make.group = vscode.TaskGroup.Build;
-        make.problemMatchers = ["$catkin-gcc"];
-
-        const test = new vscode.Task({ type: "catkin", target: "run_tests" }, "run_tests", "catkin");
-        test.execution = new vscode.ShellExecution(testCommand, {
-            env: extension.env
-        });
-        test.group = vscode.TaskGroup.Test;
-
-        return [make, test];
-    }
-
-    public resolveTask(task: vscode.Task, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.Task> {
-        return undefined;
-    }
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode"
+
+import * as extension from "../extension"
+import * as common from "./common"
+
+/**
+ * Provides catkin_make build and test tasks.
+ */
+export class CatkinMakeProvider implements vscode.TaskProvider {

```

Comment:
nit: ideally, this should be able to cover both `catkin_make` and `catkin_make_isolated`; suggest rename class to `CatkinProvider` and rename file to `catkin.ts`

Reply:
I would suggest to decide it when the pull request to support `catkin_make_isolated` is out. Despite they shared the similar names, the arguments and functionality are little bit different and it is hard to tell if we are going to use single `CatkinProvider` for them.